app/controllers/references_controller.rb

```diff
@@ -1,10 +1,37 @@
 class ReferencesController < ApplicationController
   before_action :ensure_current_user
 
+  # Shame: refactor the hell out of this
   def create
     list = List.find(params[:list_id])
-    paper = Paper.find(params[:reference][:paper_id])
-    Reference.create!(list_id: list.id, paper_id: paper.id)
+
+    # Check if uuid. If not, then ask Pubmed.
+    paper_id = params[:reference][:paper_id]
+    if uuid? paper_id
+      paper = Paper.find(paper_id)
+    else
+      pubmed = Pubmed.new
+      results = pubmed.search paper_id
+      data = results['result'][paper_id]
+      paper = (
+        Paper.find_by(pubmed_id: paper_id) ||
+        Paper.find_by(doi: paper_id) ||
+        # I really hate this due to long experience with it,
+        # but we should probably some sort of mapping
+        Paper.create(
+          pubmed_id: data['uid'],
+          title: data['title'],
+          published_at: data['pubdate'],
+          doi: data['elocationid'].sub(/^doi: /, "")
```

Comment:
I'm playing around with it and it's super cool.  I think we should add some
basic clean up of entry like removing blank spaces.  I'm also getting an
error for trying to add a doi.  So i think we should add a "No paper found"
error message.  I'll make these updates along with adding the author.

On Sun, Jul 24, 2016 at 12:00 AM, Brian Kung <notifications@github.com>
wrote:

> In app/controllers/references_controller.rb
> <https://github.com/briankung/projectcredo/pull/28#discussion_r71981890>:
>
> > +    if uuid? paper_id
> > +      paper = Paper.find(paper_id)
> > +    else
> > +      pubmed = Pubmed.new
> > +      results = pubmed.search paper_id
> > +      data = results['result'][paper_id]
> > +      paper = (
> > +        Paper.find_by(pubmed_id: paper_id) ||
> > +        Paper.find_by(doi: paper_id) ||
> > +        # I really hate this due to long experience with it,
> > +        # but we should probably some sort of mapping
> > +        Paper.create(
> > +          pubmed_id: data['uid'],
> > +          title: data['title'],
> > +          published_at: data['pubdate'],
> > +          doi: data['elocationid'].sub(/^doi: /, "")
>
> Sure. I would, but I'm stuck on the highway. Hurry. The zombies are coming.
> … <#m_1273635071504879720_>
> On Sat, Jul 23, 2016, 11:58 PM wsf246 ***@***.***> wrote: In
> app/controllers/references_controller.rb <#28 (comment)
> <https://github.com/briankung/projectcredo/pull/28#discussion_r71981864>>:
> > + if uuid? paper_id > + paper = Paper.find(paper_id) > + else > + pubmed
> = Pubmed.new > + results = pubmed.search paper_id > + data =
> results['result'][paper_id] > + paper = ( > + Paper.find_by(pubmed_id:
> paper_id) || > + Paper.find_by(doi: paper_id) || > + # I really hate this
> due to long experience with it, > + # but we should probably some sort of
> mapping > + Paper.create( > + pubmed_id: data['uid'], > + title:
> data['title'], > + published_at: data['pubdate'], > + doi:
> data['elocationid'].sub(/^doi: /, "") I can take a stab at that? — You are
> receiving this because you authored the thread. Reply to this email
> directly, view it on GitHub <
> https://github.com/briankung/projectcredo/pull/28/files/f17fd0bc5346d3be6254486f0414e4c4d0e6407b#r71981864>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ACtGxypCv-wUW_6G1gqmY6q-Cys37-csks5qYvDLgaJpZM4JTTYA>
> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/28/files/f17fd0bc5346d3be6254486f0414e4c4d0e6407b#r71981890>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZGbLTHUO4M8eKp6H8n_n5L7bTEL5ks5qYvGHgaJpZM4JTTYA>
> .
>